src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +32,11 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] newAddress = address.split(", ");
+        this.block = new Block(newAddress[0]);
```

Comment:
It should be better to split by comma and trim the space later, since user might often forget to put the space after comma.